chesster.js

```diff
@@ -1401,3 +1427,179 @@ function result_reply_missing_pairing(bot, message){
+
+//given a gamelink_id, use the lichess api to get the game details
+//pass the details to the callback as a JSON object
+function fetch_game_details(gamelink_id, callback){
+    const http = require('http');
+    var url = "http://en.lichess.org/api/game/" + gamelink_id;
+    http.get(url, (res) => {
+        var body = "";
+        res.on('data', function (chunk) {
+            body += chunk;
+        });
+        res.on('end', () => {
+            if(body != ""){
+                callback(JSON.parse(body));
+            }else{
+                callback();
+            }
+        });
+    }).on('error', (e) => {
+        console.log(e);
+        callback()
+    });
+}
+
+//verify the game meets the specified parameters in options
+function validate_game_details(details, options){
+    var result = {
+        valid: true,
+        reason: "",
+    };
+    if(details.rated != options.rated){
+        //the game is not rated correctly
+        result.valid = false;
+        result.reason = "the game is " + ( options.rated ? "unrated." : "rated." );
+    }else if( !details.clock || ( // no clock - unlimited or coorespondence
+        details.clock && ( //clock
+            details.clock.initial != options.clock.initial * 60 || // initial time
+            details.clock.increment != options.clock.increment ) // increment
+        ) 
+    ){
+        //the time control does not match options
+        result.valid = false;
+        result.reason = "the time control is incorrect."
+    }else if(details.variant != options.variant){
+        //the variant does not match
+        result.valid = false;
+        result.reason = "the variant should be standard."
+    }else{
+        //the link is too old or too new
+        var extrema = spreadsheets.get_round_extrema(options);
+        var game_start = new Date(details.timestamp);
+        if(game_start < extrema.start || game_start > extrema.end){
+            result.valid = false;
+            result.reason = "the game was not played in the current round.";
+        }
+    }
+    return result;
+}
+
+function gamelink_reply_invalid(bot, message, reason){
+    bot.reply(message, "I am sorry, <@" + message.user + ">,  "
+                     + "the link you posted is *not valid* becuase "
+                     + "*" + reason + "*");
+    bot.reply(message, "If this was a mistake, please correct it and "
+                     + "try again. If intentional, please contact one "
+                     + "of the moderators for review. Thank you.");
+}
+
+function reply_generic_failure(bot, message, contact){
+    bot.reply(message, "Something went wrong. Notify " + contact);
+}
+
+function gamelink_reply_unknown(bot, message){
+    bot.reply(message, "Sorry, I could not find that game. Please verify your gamelink.");
+}
+
+function process_gamelink(bot, message, gamelink, options){
+    //get the gamelink id if one is in the message
+    var result = spreadsheets.parse_gamelink(gamelink, options);
```

Comment:
parse_gamelink does not use the options as of yet. Yet you are passing it here. Is that an oversight, or are you planning to make it use the options in the future?